benchexec/tools/brick.py

```diff
@@ -10,22 +10,28 @@
 import benchexec.result as result
 
 
-class Tool(benchexec.tools.template.BaseTool):
+class Tool(benchexec.tools.template.BaseTool2):
     """
     Tool info for BRICK
     https://github.com/brick-tool-dev/brick-tool
     """
 
     REQUIRED_PATHS = ["bin", "lib"]
 
-    def executable(self):
-        return util.find_executable("bin/brick")
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("brick", subdir="bin")
 
     def name(self):
         return "BRICK"
 
-    def cmdline(self, executable, options, tasks, propertyfile, rlimits):
-        return [executable] + options + tasks
+    def cmdline(self, executable, options, task, rlimits):
+        data_model_param = util.get_data_model_from_task(
+            task, {"ILP32": "--32", "LP64": "--64"}
+        )
+        if data_model_param and data_model_param not in options:
+            options += [data_model_param]
+
+        return [executable] + options + list(task.input_files_or_identifier)
```

Comment:
This is quite generic. Does the tool actually support tasks with more than one input file or no input file? If not, it would be much better for user experience if the tool-info module produced a proper error in such cases by using `task.input_files` or `task.single_input_file` as appropriate.

Reply:
I have no idea if this tool or other tools with similar changes support single or multiple files . I am not a developer of this tool.
I did the changes that were upgrade the TI module without going into tool specific details.

IMO the change you suggest should be done by the tool developers.

Reply:
Note that there are cases where the tool-info module already makes it clear that only some of these cases are supported (with an `assert` or maybe in comments). In this case the code should be cleaned up.

Reply:
fixed in 6b526edc4561f9c86b55b7b52dbca0621a9c287b